### ISSUE TITLE
From() should also take channel as input

### ISSUE THREAD

**Issue by yookoala** [ahmetalpbalkan/go-linq#25]
Instead of taking an array / a slice as input, `From()` should also take a channel as input. It would provide far more flexibility.

For example,

```go
import . "github.com/ahmetalpbalkan/go-linq"

type Student struct {
    id, age int
    name string
}

func example_usage() {
    ch := make(<-chan Student)
    DBStudentToChan(ch) // some function that outputs to the channel

    over18Names, err := From(ch)
        .Where(func (s T) (bool,error){
            return s.(*Student).age >= 18, nil
        })
        .Select(func (s T) (T,error){
            return s.(*Student).name, nil
        }).Results()
}
```

**Comment by ahmetalpbalkan** [comment]
@yookoala this essentially changes the entire design of the library. Although this library is called LINQ, it's not actually LINQ as in .NET sense (which does iterators/generators –similar to channels). This library is just a slice/map manipulation utility today.

It's in a corner of my mind to create a “v2” of this library that works entirely with channels. That way we could potentially enable “data pipelines” and that would very likely enable reactive programming paradigm ([Rx](https://msdn.microsoft.com/en-us/data/gg577609.aspx)) in Go.

So for now I'd like to call it “wontfix” but we may, someday. I'm open to contributions to redo the library with this idea as well.

**Comment by ahmetalpbalkan** [comment]
@yookoala we recently merged a v2.0 and it accepts channels.
